piecrust/src/session.rs

```diff
@@ -774,6 +762,42 @@ impl Session {
     }
 }
 
+/// The receipt given for a call execution using one of either [`call`] or
+/// [`call_raw`].
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct CallReceipt<T> {
```

Comment:
For future reference, if all ICCs were to issue a `CallReceipt`, it would make sense to add a `ContractId` to this structure. 

Searching for metadata on events for a given contract would be a bit easier.